src/generators/ParticleGenerator.cpp

```diff
@@ -0,0 +1,37 @@
+
+#include "ParticleGenerator.h"
+#include <spdlog/spdlog.h>
+
+ParticleGenerator::ParticleGenerator(ParticleContainer& container)
+    : container(container)
+    , clusters(std::vector<std::unique_ptr<ParticleCluster>> {})
+{
+}
+
+void ParticleGenerator::registerCluster(std::unique_ptr<ParticleCluster> cluster)
+{
+    clusters.push_back(std::move(cluster));
+    spdlog::info("Registered cluster: {}", clusters.back()->toString());
+}
+
```

Comment:
Using the `std::unique_pointer` ensures that each cluster gets registered once. Clever! :)